src/actions_ls.rs

```diff
@@ -212,7 +228,7 @@ impl ActionHandler {
         let vfs: &Vfs = &self.vfs;
         let result: Vec<CompletionItem> = panic::catch_unwind(move || {
             let pos = adjust_vscode_pos_for_racer(params.position);
-            let file_path = &params.textDocument.file_name();
+            let file_path = &uri_string_to_file_name(&params.text_document.uri);
```

Comment:
Hum, I think the code dup here is minimal, insignificant almost. If anything, what was nicer about the original version was that it was an method of the LSP type, instead of a free function.

I think maybe we could move it back to a method if we introduce a dependency on the URL crate in `languageserver-types`, @marwes what do you think of that? I think any LS would end up needing URL/URL parsing anyways, so it seems ok to me.

The rustls code would then be like this:
```rust
let file_path = &params.text_document.file_name().unwrap();
```